zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/client/RealmAwareZkClient.java

```diff
@@ -69,12 +69,44 @@
   int DEFAULT_SESSION_TIMEOUT = 30 * 1000;
 
   // listener subscription
+  /**
+   * Subscribe the path and the listener will handle child events of the path.
+   * Add exists watch to path if the path does not exist in ZooKeeper server.
+   * WARNING: if the path is created after deletion, users need to re-subscribe the path
+   * @param path The zookeeper path
+   * @param listener Instance of {@link IZkDataListener}
+   */
   List<String> subscribeChildChanges(String path, IZkChildListener listener);
 
+  /**
+   * Subscribe the path and the listener will handle child events of the path
+   * WARNING: if the path is created after deletion, users need to re-subscribe the path
+   * @param path The zookeeper path
+   * @param listener Instance of {@link IZkDataListener}
+   * @param skipWatchingNodeNotExist Ture means not installing any watch if path does not exist.
+   */
+  List<String> subscribeChildChanges(String path, IZkChildListener listener, boolean skipWatchingNodeNotExist);
```

Comment:
Good point. Will add this to the comment. By the way, the return value semantic is still. the same. No change here. Add comment would make it explicit for people to understand.

Reply:
"No change here." not exactly. The previous interface will always add watcher, right? But the new method may not. We need to think about how to return the result so the caller does not need to guess.

Reply:
Agree. We need to give user a signal of installation failure or not. In addition, return null, or empty list. I though about it before. In Java, without reference of boolean it can be a little bit complicated. 

Let us talk about it tomorrow. 

Here is a proposal:

```java

List<String> subscribeChildChanges(String path, IZkChildListener listener, boolean skipWatchingNodeNotExist, Boolean IsInstalled);

```
isInstalled is an out parameter and would be set in the implementation to tell if installation successful or not.

Reply:
new class of result { list<string>, boolean installation type}